test/testComplexExchanger.js

```diff
@@ -0,0 +1,780 @@
+                let _oracleData = await exchanger.getOracleData.call(i);
+                assert.isArray(_oracleData, "the returned oracle data must be an array");
+                assert.lengthOf(_oracleData, 7, "there must be exact 7 items in the returned oracle data");
+                let [_oracleAddress, _oracleName, _oracleType, _waitQuery, _updateTime, _callbackTime, _rate] = _oracleData;
+                assert.equal(_oracle, _oracleAddress, "the oracle address got by oracles(i) should be equal to the one from getOracleData(i)");
+                assert.isTrue(web3.isAddress(_oracle),"the oracle address must be valid");
+                assert.equal((_oracleName.length - "0x".length) / 2, 32, "oracle name must be bytes32");
+                assert.equal((_oracleType.length - "0x".length) / 2, 16, "oracle type must be bytes16");
+                assert.equal(_waitQuery, false, "the new oracle shouldn't be waiting");
+                assert.equal(_updateTime.toNumber(), 0, "the new oracle's update time should be 0");
+                assert.equal(_callbackTime.toNumber(), 0, "the new oracle's update time should be 0");
+                assert.equal(_rate.toNumber(), 0, "the new oracle's rate should be 0");
+            }
+        });
+
+        /*
+        it("check LOCKED", async function() {
+            let exchanger = await ComplexExchanger.deployed();
+            let deadline = + await exchanger.deadline.call();
+
+            jump = deadline - now();
+            console.log(now());
+            await timeMachine.jump(jump);
+            console.log(now());
+            let state = + await exchanger.getState.call();
+            //timeMachine.setTimestamp(now() - jump);
+            //await timeMachine.jump(-jump);
+            //await acc1.sendTransaction({from: acc2, value: web3.toWei(5,'ether')});
+            //reverter.revert();
+            console.log(now());
+            assert.equal(state, StateENUM.LOCKED,"Don't correct state!!");
+        });
+        */
+
+        it("(2) check PROCESSING_ORDERS", async function() {
+            let 
+                exchanger = await ComplexExchanger.deployed(),
+                state = + await exchanger.getState.call(),
+                calcTime = + await exchanger.calcTime.call();
+
+            if (now() - calcTime > exConfig.RATE_PERIOD) {
+                assert.notEqual(state, StateENUM.PROCESSING_ORDERS, 
+                    "Don't correct state when calcTime > RATE_PERIOD");
+                await exchanger.requestRates({value: web3.toWei(5,'ether')});
+                await exchanger.calcRates();
+                state = + await exchanger.getState.call();
+            }
+
+            assert.equal(state, StateENUM.PROCESSING_ORDERS,"Don't correct state when calcTime < RATE_PERIOD");
+        });
+
+        it("(3) check WAIT_ORACLES", async function() {
+            let exchanger = await ComplexExchanger.deployed();
```

Comment:
does we need to check it on every test? check docs pls

Reply:
ne ponal))